deeppavlov/core/common/params.py

```diff
@@ -95,10 +99,15 @@ def from_params(params: Dict, mode: str = 'infer', **kwargs) -> Component:
             kwargs['mode'] = mode
 
         component = cls(**dict(config_params, **kwargs))
+        try:
+            _refs[config_params['component_name']] = component
```

Comment:
Is there a guarantee that `component_name` and `id` will be different? Are you sure that they should be written in the same structure?

Reply:
These parts of the code are no longer relevant. I deleted them.